lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -988,6 +1025,127 @@ def upload_enrollment_report(_xmodule_instance_args, _entry_id, course_id, _task
+    true_enrollment_count = 0
+    for user in enrolled_users:
+        if not user.is_staff and not CourseAccessRole.objects.filter(
+                user=user, course_id=course_id, role__in=FILTERED_OUT_ROLES
+        ).exists():
+            true_enrollment_count += 1
+
+    task_progress = TaskProgress(action_name, true_enrollment_count, start_time)
+
+    fmt = u'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
+    task_info_string = fmt.format(
+        task_id=_xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None,
+        entry_id=_entry_id,
+        course_id=course_id,
+        task_input=_task_input
+    )
+    TASK_LOG.info(u'%s, Task type: %s, Starting task execution', task_info_string, action_name)
+
+    single_purchase_total = PaidCourseRegistration.get_total_amount_of_purchased_item(course_id)
+    bulk_purchase_total = CourseRegCodeItem.get_total_amount_of_purchased_item(course_id)
+    paid_invoices_total = InvoiceTransaction.get_total_amount_of_paid_course_invoices(course_id)
+    gross_revenue = single_purchase_total + bulk_purchase_total + paid_invoices_total
+
+    all_invoices_total = Invoice.get_invoice_total_amount_for_course(course_id)
+    gross_pending_revenue = all_invoices_total - float(paid_invoices_total)
+
+    avg_price_paid = 0
+    # Avoid division by zero.
+    if true_enrollment_count != 0:
+        avg_price_paid = gross_revenue / true_enrollment_count
+
+    refunded_paid_enrollments = PaidCourseRegistration.get_paid_course_enrollment_count(
+        course_id, status='refunded'
+    )
+    refunded_reg_code_enrollments = CourseRegCodeItem.get_course_regcode_enrollment_count(
+        course_id, status='refunded'
+    )
+    total_enrollments_refunded = refunded_paid_enrollments + refunded_reg_code_enrollments
+
+    paid_courses_refunded_amount = PaidCourseRegistration.get_total_amount_of_purchased_item(
+        course_id,
+        status='refunded'
+    )
+    course_code_refunded_amount = CourseRegCodeItem.get_total_amount_of_purchased_item(course_id, status='refunded')
+    total_amount_refunded = paid_courses_refunded_amount + course_code_refunded_amount
+
+    top_discounted_codes = CouponRedemption.get_top_discount_codes_used(course_id)
+    total_coupon_codes_purchases = CouponRedemption.get_total_coupon_code_purchases(course_id)
+
+    bulk_purchases_codes = CourseRegistrationCode.order_generated_registration_codes(course_id)
+    unused_registration_codes = 0
+    for registration_code in bulk_purchases_codes:
+        if not RegistrationCodeRedemption.is_registration_code_redeemed(registration_code):
+            unused_registration_codes += 1
+
+    self_purchase_enrollment_count = PaidCourseRegistration.get_paid_course_enrollment_count(course_id)
+    bulk_purchase_enrollment_count = CourseRegCodeItem.get_course_regcode_enrollment_count(course_id)
```

Comment:
Sorry I didn't catch this last night (I didn't try bulk purchases). I think here we want is simply 'bulk_purchased_codes' which are the number of seats purchased. For example, I did one purchase of 4 seats but on my report this came back as 1 since there was just one order item. It should be 4 since there are 4 seats. Thanks.

Also, I would change some of the variable names here to be:

'self_purchased_seat_count'
'bulk_purchased_seat_count'